### DISCUSSION TITLE
How to unregister a previously registered scalar function?

### DISCUSSION THREAD

**Discussion by prashanthellina** [duckdb/duckdb#14227] [Ideas]
I am using the C-API and `duckdb_register_scalar_function` to register a scalar function. However, I don't see an API function that can unregister the registered function.

I need this to cleanly revert changes I make to a DuckDB database instance.

**Comment by Tishj** [comment]
Have you tried `drop function <function_name>` ?
I'm not sure if that hits it, but I do think an unregister function should be made regardless

**Comment by prashanthellina** [comment]
I haven't but will do so now. Thank you for the suggestion.

**Comment by prashanthellina** [comment]
I registered a function called `myfunction` and tried to drop using `DROP FUNCTION myfunction`. I get this error -> "Catalog Error: Cannot drop internal catalog entry "myfunction"!"

**Reply by Tishj** [reply]
Thanks, definitely needs a function then

**Reply by prashanthellina** [reply]
@Tishj Thank you. I moved this discussion from Q&A to Ideas. Will leave this open.
